client-app/src/common/models/search/product-search-criteria.ts

```diff
@@ -0,0 +1,23 @@
+import { KeywordSearchCriteria } from './keyword-search-criteria';
+import { ProductSearchQuery } from './product-search-query';
+import { Terms } from './terms';
+
+export class ProductSearchCriteria extends KeywordSearchCriteria {
```

Comment:
Why we can not use NSwag autogenerated class?

Reply:
Because of inheritance. But I forgot typescript has a little bit more relax algorithm of inheritance checking, so now I changed code to use generated class. SearchCriteria and KeywordSearchCriteria is also changed to be compatible.

Please review again.